crates/robonode-client/src/enroll.rs

```diff
@@ -1,17 +1,28 @@
 //! Client API for the Humanode's Bioauth Robonode.
 
 use reqwest::StatusCode;
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 
-use crate::{error_response::ErrorResponse, Client, Error};
+use crate::{error_response::ErrorResponse, Client, Error, ScanResultBlob};
 
 impl Client {
     /// Perform the enroll call to the server.
-    pub async fn enroll(&self, req: EnrollRequest<'_>) -> Result<(), Error<EnrollError>> {
+    pub async fn enroll(
+        &self,
+        req: EnrollRequest<'_>,
+    ) -> Result<EnrollResponse, Error<EnrollError>> {
         let url = format!("{}/enroll", self.base_url);
         let res = self.reqwest.post(url).json(&req).send().await?;
         match res.status() {
-            StatusCode::CREATED => Ok(()),
+            StatusCode::CREATED => {
+                if let Some(0) = res.content_length() {
+                    Ok(EnrollResponse {
+                        scan_result_blob: None,
+                    })
+                } else {
+                    Ok(res.json().await?)
+                }
+            }
```

Comment:
```suggestion
            StatusCode::CREATED if let Some(0) = res.content_length() => Ok(EnrollResponse {
                    scan_result_blob: None,
            })
            StatusCode::CREATED => Ok(res.json().await?),
```

A slightly more compact way to represent the same logic